src/database.rs

```diff
@@ -332,45 +340,74 @@ pub struct ModifyOrder {
 
 pub struct MarkPaid {
     pub order: String,
-    pub res: oneshot::Sender<Result<OrderInfo, DbError>>,
+    pub res: oneshot::Sender<Result<OrderDb, DbError>>,
+}
+
+#[derive(Encode, Decode)]
+pub struct OrderDb {
+    pub inner: OrderInfo,
+    pub start: u64,
+    pub death: u64,
 }
 
 fn create_order(
     order: String,
-    order_info: OrderInfo,
+    order_inner: OrderInfo,
     orders: &sled::Tree,
+    account_lifetime: u64,
 ) -> Result<OrderCreateResponse, DbError> {
-    Ok(match orders.get(&order)? {
-        Some(record) => {
-            let old_order_info = OrderInfo::decode(&mut &record[..])?;
-            match order_info.payment_status {
-                PaymentStatus::Pending => {
-                    drop(orders.insert(order.encode(), order_info.encode())?);
-                    OrderCreateResponse::Modified
-                }
-                PaymentStatus::Paid => OrderCreateResponse::Collision(old_order_info),
+    Ok(if let Some(record) = orders.get(&order)? {
+        let old_order_info = OrderDb::decode(&mut &record[..])?;
+        match order_inner.payment_status {
+            PaymentStatus::Pending => {
+                drop(
+                    orders.insert(
+                        order.encode(),
+                        OrderDb {
+                            inner: order_inner,
+                            start: old_order_info.start,
```

Comment:
this is an interesting question; I'd assume we should restart lifetime on modification because this is really a completely different order now.

What a misleading feature.